contrib/mesos/pkg/scheduler/podtask/pod_task.go

```diff
@@ -213,35 +284,41 @@ func (t *T) AcceptOffer(offer *mesos.Offer) bool {
 	if offer == nil {
 		return false
 	}
+
+	// check ports
+	if _, err := t.mapper.Generate(t, offer); err != nil {
+		log.V(3).Info(err)
+		return false
+	}
+
+	// find offered cpu and mem
 	var (
-		cpus float64 = 0
-		mem  float64 = 0
+		offeredCpus CPUShares
+		offeredMem  MegaBytes
 	)
 	for _, resource := range offer.Resources {
 		if resource.GetName() == "cpus" {
-			cpus = *resource.GetScalar().Value
+			offeredCpus = CPUShares(*resource.GetScalar().Value)
 		}
 
 		if resource.GetName() == "mem" {
-			mem = *resource.GetScalar().Value
+			offeredMem = MegaBytes(*resource.GetScalar().Value)
 		}
 	}
-	if _, err := t.mapper.Generate(t, offer); err != nil {
-		log.V(3).Info(err)
-		return false
-	}
 
-	// for now hard-coded, constant values are used for cpus and mem. This is necessary
-	// until parent-cgroup integration is finished for mesos and k8sm. Then the k8sm
-	// executor can become the parent of pods and subsume their resource usage and
-	// therefore be compliant with expectations of mesos executors w/ respect to
-	// resource allocation and management.
-	//
-	// TODO(jdef): remove hardcoded values and make use of actual pod resource settings
-	if (cpus < DefaultContainerCpus) || (mem < DefaultContainerMem) {
-		log.V(3).Infof("not enough resources: cpus: %f mem: %f", cpus, mem)
+	// calculate cpu and mem sum over all containers of the pod
+	// TODO (@sttts): also support pod.spec.resources.limit.request
+	// TODO (@sttts): take into account the executor resources
+	podCopy := t.Pod // make copy to fill in the limits without modifying the pod
```

Comment:
alternatively, don't make the change here; there's already an attempt to copy the pod object (or at least parts of it) in the scheduler (near where annotations are set, I think)